working-groups/unsafe-code-guidelines/NOTES.md

```diff
@@ -1,3 +1,89 @@
 This file will contain summaries from the meetings where the
 unsafe-code-guidelines WG synchronizes with the lang team and broader
 community. That hasn't happened yet. =)
+
+# First meeting agenda (Date TBD)
+
+## Review of how UCG has operated thus far
+
+- The UCG reference lives on the repository
+- At any given time we have an "active area of discussion":
+    - This is proposed in a PR which describes an initial set of discussion topics
+      ([example](https://github.com/rust-lang/unsafe-code-guidelines/pull/54)).
+    - When an issue seems to reach consensus, somebody writes up a
+      summary in the form of a PR that edits the reference
+      ([example](https://github.com/rust-lang/unsafe-code-guidelines/pull/57)).
+    - Sometimes, the consensus contains further points of contention
+      -- in this case, we open up new issues representing just those
+      points (and reference them from the reference).
+- Weekly meetings on Zulip; mostly try to look for issues that
+  seem to have reached consensus and assign work, don't usually involve
+  a lot of technical discussion.
+  
+Experience with this system? It seems to be working reasonably well,
+though we've not had as much participation as I had hoped, and we've
+been moving a bit slower than I might've thought. I think this is
+largely because these discussions are nobody's top priority right
+now. -nikomatsakis
+
+## General roadmap
+
+Our immediate goal is to produce an "RFC" covering two inter-related areas:
+
+- [**Layout:**](https://github.com/rust-lang/unsafe-code-guidelines/blob/master/active_discussion/layout.md)
+  - What are the layout rules that unsafe code authors can rely on?
+  - When do we guarantee ABI compatibility between Rust types and C types?
+  - When can you rely on "enum layout" optimizations?
+- [**Validity invariants:**](https://github.com/rust-lang/unsafe-code-guidelines/blob/master/active_discussion/validity.md)
+  - Which invariants derived from types are there that the compiler
+    expects to be always maintained, and (equivalently) that unsafe
+    code must always uphold (or else cause undefined behavior)?
+
+## Current status
+
+We've finished up the **layout** conversation. Some of the key points:
+
+- [Struct layout](https://github.com/rust-lang/unsafe-code-guidelines/blob/master/reference/src/layout/structs-and-tuples.md)
+  - The layout of tuples `(T1..Tn)` is defined "as if" there were a corresonding struct
```

Comment:
It is? Is there going to be an RFC for this? @sgrif and I have discussed lots of times that the easiest path towards variadic generics would preclude this, and I'd like to make sure we've fully evaluated that consequence.

Reply:
Exactly the sort of thing I hope to catch with these sync meetings. 

In answer to your questions: this is a WG, so any official decision would require an RFC, yes.

Regarding the interactions with variadic generics, seems like a good thing we should talk about!